### PR TITLE
Fix #1486: Creating CRDs with schema validation is broken

### LINKED ISSUE #1486: Creating CRDs with schema validation is broken
the K8s api server makes a lot of strict checks, for instance it fails if the schema contains default values, definitions, $ref elements, etc. One of the checks is:

```go
	if schema.Dependencies != nil {
		for dependency, jsonSchemaPropsOrStringArray := range schema.Dependencies {
			allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISchema(jsonSchemaPropsOrStringArray.Schema, fldPath.Child("dependencies").Key(dependency), ssv)...)
		}
	}
```

source: https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/validation/validation.go#L671

However, even if the `JSONSchemaProps` object has the `dependecies` field set to `null`, the object mapper from Jackson that's being used in [here](https://github.com/fabric8io/kubernetes-client/blob/a50f3e946f20b298f69f065a2597fc94a0cb2743/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationSupport.java#L230) converts the `null` to an empty `LinkedHashMap` and this fails on the K8s api server. So there is no way currently to create the schema validation with the fabric8 k8s client.

I am using the `.withNewOpenAPIV3SchemaLike(schema)` method for the `CustomResourceDefinitionBuilder`.

I believe there must be some configuration of the object mapper so that it shouldn't translate nulls to empty maps.

```java
JSON_MAPPER.setSerializationInclusion(Include.NON_NULL);
```
..looks promising

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #1486 

+ Upgrade to sundrio v0.19.2
+ Change sundrio configuration to disable lazy collection initialization

### CODE REVIEW

**Comment by rohanKanojia** [comment]
build seems to failing due to OutOfMemory error:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project kubernetes-model: Fatal error compiling: java.lang.OutOfMemoryError: RegularFileObject[/mnt/hudson_workspace/workspace/kubernetes-client-pullreq/kubernetes-model/kubernetes-model/target/generated-sources/annotations/io/fabric8/kubernetes/api/model/KubeSchemaFluentImpl.java]@pos236525: RegularFileObject[/mnt/hudson_workspace/workspace/kubernetes-client-pullreq/kubernetes-model/kubernetes-model/target/generated-sources/annotations/io/fabric8/kubernetes/api/model/KubeSchemaFluentImpl.java]@pos236562: RegularFileObject[/mnt/hudson_workspace/workspace/kubernetes-client-pullreq/kubernetes-model/kubernetes-model/target/generated-sources/annotations/io/fabric8/kubernetes/api/model/KubeSchemaFluentImpl.java]@pos236596: RegularFileObject[/mnt/hudson_workspace/workspace/kubernetes-client-pullreq/kubernetes-model/kubernetes-model/target/generated-sources/annotations/io/fabric8/kubernetes/api/model/KubeSchemaFluentImpl.java]@pos236620: GC overhead limit exceeded -> [Help 1]
```

Any ideas on how to increase heap size on ci.fabric8.io?

**Comment by rohanKanojia** [comment]
retest this please

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/StorageSpaceCrudTest.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
+import static junit.framework.TestCase.assertNull;
```
Comment (manusa) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/StorageSpaceCrudTest.java]: Maybe better use `org.junit.jupiter.api.Assertions.assertNull` just for consistency

**Comment by rohanKanojia** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**